src/lib.rs

```diff
@@ -106,6 +109,57 @@ pub enum ReturnIndent {
 
 impl_enum_decodable!(ReturnIndent, WithArgs, WithWhereClause);
 
+enum ErrorKind {
+    // Line has more than config!(max_width) characters
+    LineOverflow,
+    // Line ends in whitespace
+    TrailingWhitespace,
+    // TO-DO item in comments without an issue number
+    UnnumberedTodo,
+    // FIX-ME item in comments without an issue number
+    UnnumberedFixme,
+}
+
+impl<'a> fmt::Display for ErrorKind {
+    // Prints all the formatting errors.
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        let msg = match *self {
+            ErrorKind::LineOverflow => "line exceeded maximum length",
+            ErrorKind::TrailingWhitespace => "left trailing whitespace",
```

Comment:
nit the "left" here could be interpreted ambiguously (left cf right as opposed to left over)